src/components/Jam.jsx

```diff
@@ -0,0 +1,300 @@
+              <Balance loading={true} />
+            ) : (
+              // Todo: Subtract frozen or locked UTXOs from amount shown.
+              <Balance
+                valueString={walletInfo.data.display.walletinfo.total_balance}
+                convertToUnit={settings.unit}
+                showBalance={settings.showBalance}
+              />
+            )}
+          </div>
+          <p className="text-secondary mb-4">{t('schedule.description_destination_addresses')}</p>
+        </>
+      )}
+      <Formik
+        initialValues={{ dest1: '', dest2: '', dest3: '' }}
+        validate={(values) => {
+          if (collaborativeOperationRunning) {
+            return {}
+          }
+
+          const errors = {}
+
+          const isValidAddress = (candidate) => {
+            return typeof candidate !== 'undefined' && candidate !== ''
+          }
+
+          if (!isValidAddress(values.dest1)) {
+            errors.dest1 = t('schedule.error_invalid_destionation_address')
+          }
+          if (!isValidAddress(values.dest2)) {
+            errors.dest2 = t('schedule.error_invalid_destionation_address')
+          }
+          if (!isValidAddress(values.dest3)) {
+            errors.dest3 = t('schedule.error_invalid_destionation_address')
+          }
+
+          return errors
+        }}
+        onSubmit={async (values) => {
+          if (collaborativeOperationRunning) {
+            await stopSchedule()
+          } else {
+            await startSchedule(values)
+          }
+        }}
+      >
+        {({
+          values,
+          isSubmitting,
+          handleSubmit,
+          handleBlur,
+          handleChange,
+          setFieldValue,
+          isValid,
+          dirty,
+          touched,
+          errors,
+        }) => (
+          <rb.Form onSubmit={handleSubmit} noValidate>
+            {!collaborativeOperationRunning && (
+              <>
+                <rb.Form.Group className="mb-4" controlId="offertype">
+                  <ToggleSwitch
+                    label={t('schedule.toggle_internal_destination_title')}
+                    subtitle={t('schedule.toggle_internal_destination_title', { account: INTERNAL_DEST_ACCOUNT })}
+                    initialValue={destinationIsInternal}
+                    onToggle={async (isToggled) => {
+                      setDestinationIsInternal(isToggled)
+
+                      if (isToggled) {
+                        const newAddresses = getNewAddresses(3, INTERNAL_DEST_ACCOUNT)
+                        await setFieldValue('dest1', newAddresses[0], true)
+                        await setFieldValue('dest2', newAddresses[1], true)
+                        await setFieldValue('dest3', newAddresses[2], true)
+                      } else {
+                        await setFieldValue('dest1', '', false)
+                        await setFieldValue('dest2', '', false)
+                        await setFieldValue('dest3', '', false)
```

Comment:
Very good catch. I guess the issue is that without the `await`, the validate function gets triggered once for each `setFieldValue` call resulting it only seeing the field that was changed by the call that triggered the validation:

<img width="333" alt="Screenshot 2022-05-10 at 12 14 50" src="https://user-images.githubusercontent.com/10026790/167606248-26d1bb93-3e29-4e70-bf36-50ec3965739a.png">

With the `await`, the validation function magically sees the values changed by the other `setFieldValue` calls:

<img width="652" alt="Screenshot 2022-05-10 at 12 14 04" src="https://user-images.githubusercontent.com/10026790/167606269-baf90e2a-b45f-41d8-a3a3-e70ffb7596a3.png">

So I guess it's just pure luck that it works. However, not sure if we should be relying on such undocumented behavior, though. I'll have a look to see if I can find a workaround. There's an [endless thread](https://github.com/jaredpalmer/formik/issues/529) on the Formik repo about something similar with no good conclusion unfortunately.

Reply:
Uff.. sorry for getting you into that. Should we add a small warning as comment?
Otherwise I already know that I will try to remove them.. because I forget that they serve a purpose.

Reply:
Ok changed it so that the form is explicitly verified again when all 3 addresses are filled out. Might be not ideal but at least works based on documented APIs only. Let me know what you think!